src/sidebar/components/PendingUpdatesNotification.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, RefreshIcon } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import { useCallback, useEffect, useRef, useState } from 'preact/hooks';
+
+import { useShortcut } from '../../shared/shortcut';
+import { withServices } from '../service-context';
+import type { StreamerService } from '../services/streamer';
+import { useSidebarStore } from '../store';
+
+export type PendingUpdatesNotificationProps = {
+  // Injected
+  streamer: StreamerService;
+
+  // Test seams
+  setTimeout_?: typeof setTimeout;
+  clearTimeout_?: typeof clearTimeout;
+};
+
+const collapseDelay = 5000;
+
+function PendingUpdatesNotification({
+  streamer,
+  /* istanbul ignore next - test seam */
+  setTimeout_ = setTimeout,
+  /* istanbul ignore next - test seam */
+  clearTimeout_ = clearTimeout,
+}: PendingUpdatesNotificationProps) {
+  const store = useSidebarStore();
+  const pendingUpdateCount = store.pendingUpdateCount();
+  const hasPendingUpdates = store.hasPendingUpdates();
+  const applyPendingUpdates = useCallback(
+    () => streamer.applyPendingUpdates(),
+    [streamer],
+  );
+  const [showCollapsed, setShowCollapsed] = useState(false);
+  const currentTimeout = useRef<number | null>(null);
+
+  useShortcut('l', () => hasPendingUpdates && applyPendingUpdates());
+
+  useEffect(() => {
+    const clearCurrentTimeout = () =>
+      currentTimeout.current && clearTimeout_(currentTimeout.current);
+
+    if (hasPendingUpdates) {
+      clearCurrentTimeout();
```

Comment:
When this effect runs, the cleanup from the previous effect run will have always been called first, so there should never be an active timeout at this point.

Reply:
True